src/main/java/io/pinecone/clients/Inference.java

```diff
@@ -0,0 +1,94 @@
+package io.pinecone.clients;
+
+import org.openapitools.control.client.ApiClient;
+import org.openapitools.control.client.ApiException;
+import org.openapitools.control.client.api.InferenceApi;
+import org.openapitools.control.client.model.EmbedRequest;
+import org.openapitools.control.client.model.EmbedRequestInputsInner;
+import org.openapitools.control.client.model.EmbedRequestParameters;
+import org.openapitools.control.client.model.EmbeddingsList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * The Inference class provides methods to interact with Pinecone's embedding API through the Java SDK. It allows users
+ * to send input data to generate embeddings using a specified model.
+ * <p>
+ * This class utilizes the {@link InferenceApi} to make API calls to the Pinecone embedding service.
+ *
+ * <pre>{@code
+ *      import io.pinecone.clients.Inference;
+ *      import io.pinecone.clients.Pinecone;
+ *      import org.openapitools.control.client.ApiException;
+ *      import org.openapitools.control.client.model.EmbeddingsList;
+ *
+ *      import java.util.ArrayList;
+ *      import java.util.HashMap;
+ *      import java.util.List;
+ *      import java.util.Map;
+ *
+ *     // Prepare input sentences to be embedded
+ *     List<String> inputs = new ArrayList<>(1);
+ *     inputs.add("The quick brown fox jumps over the lazy dog.");
+ *     inputs.add("Lorem ipsum");
+ *
+ *     // Specify the embedding model and parameters
+ *     String embeddingModel = "multilingual-e5-large";
+ *
+ *     Map<String, Object> parameters = new HashMap<>();
+ *     parameters.put("input_type", "query");
+ *     parameters.put("truncate", "END");
```

Comment:
This syntax is a bit more concise for code samples:

```java
List<String> inputs = List.of(
    "The quick brown fox jumps over the lazy dog.",
    "Lorem ipsum"
);

Map<String, Object> parameters = Map.of(
        "input_type", "query",
        "truncate", "END"
);
```

Reply:
Fixed!

Reply:
Wait I just realized Java 8 doesn't support this, its from Java 9 and later. Will revert this change.

Reply:
It's just sample code though?

Reply:
Yes but since the SDK supports Java 8+, I would prefer all functionalities that are supported in Java 8 as well. Also its just the syntax right? Like the there will be one object instantiated regardless of the approach?